### DISCUSSION TITLE
Confirm which compiler we're using and if using clang could be faster to compile

### DISCUSSION THREAD

**Discussion by gustavovalverde** [ZcashFoundation/zebra#3679] [Ideas]
## Motivation
`clang` builds faster than `gcc` (linux default) but if we have both available, we have to ensure we're using `clang` when running `cargo build`

Origin: https://github.com/ZcashFoundation/zebra/pull/3431#discussion_r796168159

### Specifications

The C compiler should be auto-detected.

There seems to be a variable to set the preferred compiler: https://github.com/rust-lang/rust/issues/635#issuecomment-1793707
Maybe we need this too (?) https://stackoverflow.com/a/57817848

## Tasks

Try installing `clang`, and making sure `gcc g++` are not installed, and see if any parts of the build fail.

**Comment by teor2345** [comment]
The C compiler should be auto-detected.

We should try installing `clang` and not `gcc g++`, and see if any parts of the build fail.

**Comment by gustavovalverde** [comment]
At least on the Dockerfile, Rust comes with `gcc` and `g++` preinstalled. We're adding `clang`. I'm not sure if removing `gcc g++` would be the best option.

**Comment by ftm1000** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/3451?planning-poker) with ZenHub @gustavovalverde @conradoplg @dconnolly @jvff @oxarbitrage @teor2345 @upbqdn

**Comment by teor2345** [comment]
> At least on the Dockerfile, Rust comes with `gcc` and `g++` preinstalled. We're adding `clang`. I'm not sure if removing `gcc g++` would be the best option.

Ah right. Maybe we should only remove `gcc` and `g++` to check if `clang` is quicker. And if it will work for everything we build.

Then if it is quicker, we can find the right config to always use `clang` without removing `gcc`.

**Comment by mpguerra** [comment]
closing as stale
